smbios/decoder.go

```diff
@@ -189,6 +189,15 @@ func (d *Decoder) next() (*Structure, error) {
 
 		systemInfo.BaseboardInfo = bbInfo
 	}
+	if h.Type == 3 {
+
+		sysEnclosure := (*SystemEnclosure)(unsafe.Pointer(&fb[0]))
+
+		systemInfo.SystemEnclosure = &SystemEnclosure{}
+
+		systemInfo.SystemEnclosure.Type = sysEnclosure.Type & 0x7F

```

Comment:
Please place the comments as why we are using the number 0x7F, this you can copy from the PDF file.

Reply:
Fixed